--- dart/runtime/bin/bin.gypi.old	2015-05-26 00:39:55.198509235 +0200
+++ dart/runtime/bin/bin.gypi	2015-05-26 00:40:35.098001796 +0200
@@ -221,11 +221,6 @@
         ['exclude', '_test\\.(cc|h)$'],
       ],
       'conditions': [
-        ['dart_io_support==1 and dart_io_secure_socket==1', {
-          'dependencies': [
-            'bin/net/ssl.gyp:libssl_dart',
-          ],
-        }],
         ['dart_io_secure_socket==0', {
           'defines': [
             'DART_IO_SECURE_SOCKET_DISABLED'
@@ -290,11 +285,6 @@
         'io_natives.cc',
       ],
       'conditions': [
-        ['dart_io_support==1 and dart_io_secure_socket==1', {
-          'dependencies': [
-            'bin/net/ssl.gyp:libssl_dart',
-          ],
-        }],
         ['dart_io_support==1 and dart_io_secure_socket==0', {
           'dependencies': [
             'bin/net/zlib.gyp:zlib_dart',
--- dart/runtime/bin/io_impl_sources.gypi.old	2015-05-26 00:48:19.662093486 +0200
+++ dart/runtime/bin/io_impl_sources.gypi	2015-05-26 00:48:25.562018452 +0200
@@ -5,6 +5,22 @@
 # This file contains some C++ sources for the dart:io library.  The other
 # implementation files are in builtin_impl_sources.gypi.
 {
+  'include_dirs': [
+    '<(sysroot)/usr/include/nspr',
+    '<(sysroot)/usr/include/nss',
+  ],
+  'link_settings': {
+    'libraries': [
+      '-lplds4',
+      '-lplc4', 
+      '-lnspr4',
+      '-lssl3',
+      '-lsmime3',
+      '-lnss3',
+      '-lnssutil3',
+      '-lz',
+    ],
+  },
   'sources': [
     'eventhandler.cc',
     'eventhandler.h',
