--- dart/third_party/boringssl/src/crypto/bio/socket_helper.c.old	2015-12-11 02:57:53.318984528 +0100
+++ dart/third_party/boringssl/src/crypto/bio/socket_helper.c	2015-12-11 02:57:56.328939391 +0100
@@ -12,7 +12,7 @@
  * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE. */
 
-#define _POSIX_SOURCE
+#define POSIX_SOURCE
 
 #include <openssl/bio.h>
 #include <openssl/err.h>
--- dart/tools/gyp/configurations.gypi.old	2016-01-03 23:33:47.371806012 +0100
+++ dart/tools/gyp/configurations.gypi	2016-01-03 23:34:31.371152638 +0100
@@ -8,6 +8,7 @@
       '-Wall',
       '-Wextra', # Also known as -W.
       '-Wno-unused-parameter',
+      '-Wno-unused-variable',
     ],
 
     # Default value.  This may be overridden in a containing project gyp.
