--- dart/build/toolchain/get_concurrent_links.py	2021-03-26 11:31:08.546513012 +0100
+++ dart/build/toolchain/get_concurrent_links.py	2021-03-26 11:44:20.999817822 +0100
@@ -48,7 +48,7 @@
                     if not match:
                         continue
                     # Allow 8Gb per link on Linux because Gold is quite memory hungry
-                    return max(1, int(match.group(1)) / (8 * (2**20)))
+                    return max(1, int(int(match.group(1)) / (8 * (2**20))))
         return 1
     elif sys.platform == 'darwin':
         try:
@@ -64,4 +64,4 @@
         return 1
 
 
-print GetDefaultConcurrentLinks()
+print(str(GetDefaultConcurrentLinks()))
--- dart/tools/make_version.py	2021-03-26 11:29:59.432696101 +0100
+++ dart/tools/make_version.py	2021-03-26 11:30:33.000608848 +0100
@@ -85,7 +85,7 @@
         version_time = utils.GetGitTimestamp()
     if version_time == None:
         version_time = 'Unknown timestamp'
-    version = version.replace('{{COMMIT_TIME}}', version_time.decode('utf-8'))
+    version = version.replace('{{COMMIT_TIME}}', version_time)
 
     snapshot_hash = MakeSnapshotHashString()
     version = version.replace('{{SNAPSHOT_HASH}}', snapshot_hash)
