--- dart/tools/gyp/configurations_make.gypi.old	2015-09-05 03:46:15.166329619 +0200
+++ dart/tools/gyp/configurations_make.gypi	2015-09-05 03:47:06.865670755 +0200
@@ -10,9 +10,6 @@
     'configurations': {
       'Dart_Linux_Base': {
         'abstract': 1,
-        'defines': [
-          '_FORTIFY_SOURCE=2',
-        ],
         'cflags': [
           '-Werror',
           '<@(common_gcc_warning_flags)',
