--- a/IGC/BiFModule/CMakeLists.txt      2019-03-21 13:53:58.000000000 +0100
+++ b/IGC/BiFModule/CMakeLists.txt      2019-03-27 19:27:42.802909652 +0100
@@ -252,9 +252,9 @@
   endif()
   add_dependencies(common_clang_dll GetClang)
 
-  if(UNIX)
-    install(FILES $<TARGET_FILE:common_clang_dll> DESTINATION ${IGC_INSTALL_TIME_ROOT_DIR}/lib COMPONENT igc-opencl)
-  endif(UNIX)
+#  if(UNIX)
+#    install(FILES $<TARGET_FILE:common_clang_dll> DESTINATION ${IGC_INSTALL_TIME_ROOT_DIR}/lib COMPONENT igc-opencl)
+#  endif(UNIX)
 
 # End Clang section
 
--- a/IGC/CMakeLists.txt	2019-03-21 13:53:58.000000000 +0100
+++ b/IGC/CMakeLists.txt	2019-03-27 09:23:14.341157852 +0100
@@ -3250,6 +3250,7 @@
     "LLVMCore"
     "LLVMSupport"
     "LLVMDemangle"
+    "LLVMBinaryFormat"
     )
 	
 
@@ -3725,7 +3726,8 @@
       ${IGC_BUILD__LLVM_LIBS_TO_LINK}
       "${IGC_BUILD__END_GROUP}"
     )
-
+  list(APPEND _targetLinkLine "dl")
+  
   if(_targetIsStatic)
     # Link line for projects that use IGC as static library (allows to resolve multi-circular dependencies between libraries).
     set("IGC_BUILD__LINK_LINE${_libBuildSuffix}"
@@ -3782,6 +3784,7 @@
       ${IGC_BUILD__LLVM_LIBS_TO_LINK}
       "${IGC_BUILD__END_GROUP}"
     )
+  list(APPEND _targetLinkLine "dl")
 
 
 list(REMOVE_ITEM _targetLinkLine
@@ -3874,8 +3877,8 @@
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/AdaptorOCL/igc-opencl.pc.in ${IGC_BUILD__IGC_BIN_DIR}/AdaptorOCL/igc-opencl.pc @ONLY)
 
 if(UNIX)
-  install(TARGETS ${IGC_BUILD__PROJ__igc_dll} DESTINATION ${IGC_INSTALL_TIME_ROOT_DIR}/lib COMPONENT igc-core)
-  install(TARGETS ${IGC_BUILD__PROJ__fcl_dll} DESTINATION ${IGC_INSTALL_TIME_ROOT_DIR}/lib COMPONENT igc-opencl)
+  install(TARGETS ${IGC_BUILD__PROJ__igc_dll} LIBRARY COMPONENT igc-core)
+  install(TARGETS ${IGC_BUILD__PROJ__fcl_dll} LIBRARY COMPONENT igc-opencl)
   install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/AdaptorOCL/ocl_igc_shared DESTINATION ${IGC_INSTALL_TIME_ROOT_DIR}/include/igc COMPONENT igc-opencl-devel)
   install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/AdaptorOCL/ocl_igc_interface DESTINATION ${IGC_INSTALL_TIME_ROOT_DIR}/include/igc COMPONENT igc-opencl-devel)
   install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/AdaptorOCL/cif DESTINATION ${IGC_INSTALL_TIME_ROOT_DIR}/include/igc COMPONENT igc-opencl-devel)
--- a/visa/iga/IGALibrary/CMakeLists.txt        2019-03-21 13:53:58.000000000 +0100
+++ b/visa/iga/IGALibrary/CMakeLists.txt        2019-03-27 19:12:10.894353495 +0100
@@ -334,9 +334,7 @@
 
   if(UNIX)
     INSTALL(TARGETS IGA_DLL
-        RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/.   COMPONENT igc-core
-        LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib COMPONENT igc-core
-        ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/.   COMPONENT igc-core
+        LIBRARY   COMPONENT igc-core
     )
   elseif(WIN32)
     install(TARGETS IGA_DLL RUNTIME
